pkg/result/processors/exclusion_generated_file_matcher.go

```diff
@@ -0,0 +1,107 @@
+package processors
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"strings"
+
+	"github.com/golangci/golangci-lint/pkg/config"
+	"github.com/golangci/golangci-lint/pkg/logutils"
+)
+
+// The values must be in lowercase.
+const (
+	genCodeGenerated = "code generated"
+	genDoNotEdit     = "do not edit"
+
+	// Related to easyjson.
+	genAutoFile = "autogenerated file"
+
+	//nolint:lll // Long URL
+	// Related to Swagger Codegen.
+	// https://github.com/swagger-api/swagger-codegen/blob/61cfeac3b9d855b4eb8bffa0d118bece117bcb7d/modules/swagger-codegen/src/main/resources/go/partial_header.mustache#L16
+	// https://github.com/swagger-api/swagger-codegen/issues/12358
+	genSwaggerCodegen = "* generated by: swagger codegen "
+)
+
+// GeneratedFileMatcher detects generated files.
+//   - mode "lax": see `isGeneratedFileLax` documentation.
+//   - mode "strict": see `isGeneratedFileStrict` documentation.
+//   - mode "disable": skips this processor.
+type GeneratedFileMatcher struct {
+	debugf logutils.DebugFunc
+
+	mode string
+}
+
+func NewGeneratedFileMatcher(mode string) *GeneratedFileMatcher {
+	return &GeneratedFileMatcher{
+		debugf: logutils.Debug(logutils.DebugKeyGeneratedFileFilter),
+		mode:   mode,
+	}
+}
+
+func (p *GeneratedFileMatcher) IsGeneratedFile(filepath string, src any) (bool, error) {
+	if p.mode == config.GeneratedModeDisable {
+		return false, nil
+	}
+
+	file, err := parser.ParseFile(token.NewFileSet(), filepath, src, parser.PackageClauseOnly|parser.ParseComments)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse file: %w", err)
+	}
+
+	if p.mode == config.GeneratedModeStrict {
+		return isGeneratedFileStrict(file), nil
+	}
+
+	doc := getComments(file)
+
+	return p.isGeneratedFileLax(doc), nil
+}
+
+// isGeneratedFileLax reports whether the source file is generated code.
+// The function uses a bit laxer rules than isGeneratedFileStrict to match more generated code.
+// See https://github.com/golangci/golangci-lint/issues/48 and https://github.com/golangci/golangci-lint/issues/72.
```

Comment:
Can be simple:
```suggestion
// See #48 and #72.
```

Reply:
The `#` links are not rendering in IDEs.